### ISSUE TITLE
Types with 'private' or 'protected' accessibility cannot be used with log methods

### ISSUE THREAD

**Issue by bekir-ozturk** [bekir-ozturk/flying-logs#2]
Today, log methods are all living under the 'Log' type. They don't have visibility into private or protected types, which prevents them from being used in log methods. Consider the case:

```csharp
public class Book
{
    private State _state;

    public void Borrow()
    {
        Log.Information.L83("Borrowing a {state} book.", _state); <---- Call site where type of '_state' is known.
    }

    private enum State
    {
        BrandNew,
        Used
    }
}


// ------------------------------------

internal class Log
{
    internal class Information
    {
        public void L83(string template, Book.State state)  <--- Compilation error. 'State' is inaccessible due to its protection level.
        {
            // ...
        }
    }
}
```

Thinking as I'm typing:
We can't move the log method under the `Book` type as it wouldn't be accessible via `Log.Information....` syntax even if the type was marked `partial`. Although it sounds super hacky, we could define another `Log` type under `Book` to avoid the change in syntax. But that means having both `FlyingLogs.Log` and `Book.Log` in the same context and it will cause ambiguity that the developer will need to resolve.

Still, there is no other way to access such types. This might be our best option.
An alternative is to emit an error and do not provide support for these cases.

Investigate.
